src/compiler/transformers/es2017.ts

```diff
@@ -83,6 +85,103 @@ namespace ts {
             }
         }
 
+        function asyncBodyVisitor(node: Node): VisitResult<Node> {

```

Comment:
Is there a canonical source we can reference in a comment here for the list-of-things-which-may-contain-or-whose-children-may-contain-hoisted-declarations? It seems nonobvious which elements of [the statement and declaration list](https://tc39.github.io/ecma262/#sec-ecmascript-language-statements-and-declarations) need to be explicitly visited here.

Also: Does `LabeledStatement` need to be handled? eg `foo: {
  var a;
}`

Reply:
Good catch

Reply:
I've added `isNodeWithPossibleVarDeclaration` in 2ba29d8 as a way to verify this list.